### PR TITLE
T2I Reproducibility: Support end-to-end English-German pipeline runs

### PR DESCRIPTION
**Pull request by anthonymirand**
After multiple rounds of trial and error, this PR supports preliminary end-to-end runs of the T2I English-German pipeline (woo-ish!)

Summary of noteworthy changes 
- Merge the `main` into the existing `add_text_input_ec` branch (the only merge conflicts were in `EC_finetune/agents.py` and `EC_finetune/senders.py`): [merge commit](https://github.com/CLMBRs/communication-translation/commit/b35d93cfdb9a37d7116be86587fbf5387e9d78d0)
- Move T2I config files to `Configs/t2i/` and add necessary RunScripts for the T2I en-de pipeline
- Add back `set_seed` to `EC_finetune/__main__.py`
- Add comment explaining why the current command line arguments + config loading is bug prone 

TODO
- Use Hydra instead of merging command line arguments and config files
- Add T2I support for the rest of the supported languages

| Source  | en->de  | de->en  |
|---|---|---|
| Paper  |  18.26  | 25.77   | 
| Patas  |  17.1855 | 24.1958  |

### CODE REVIEW

File: EC_finetune/__main__.py [hunk 1]
```diff
@@ -428,12 +458,31 @@ def main():
...
+                args,
+                max_length=128
+            )
+            valid_set = TextInputECDataset(
+                valid_images,
+                valid_captions,
+                args.num_distractors_valid,
+                tokenizer,
+                args,
+                max_length=128,
```
Comment (shanest) [inline EC_finetune/__main__.py:476]: These two `max_length` shouldn't be hard-coded, but a default arg elsewhere

File: EC_finetune/senders.py [hunk 2]
```diff
@@ -132,40 +141,75 @@ def forward(
...
+                    sender_image.device
+                )
+                for i in range(self.unroll_length):
+                    next_hidden, (h, c) = self.lstm(sender_image, (h, c))
+                    unrolled_hidden.append(next_hidden)
+                    sender_image = next_hidden
+                sender_image = torch.stack(unrolled_hidden, dim=1).squeeze()
+            elif self.unroll == 'transformer':
+                sender_image = self.transformer(sender_image)
+            crossattention_input = sender_image
```
Comment (shanest) [inline EC_finetune/senders.py:194]: Great work here!  There's probably a uniform API here for an `unroller`, and `self.unroller`, where the LSTM vs transformer logic happens elsewhere, but I just want to flag that as something to think about for the future; doesn't need to be added to this PR

**Review by shanest — COMMENTED** [review]
Looks great to me!  Just want to confirm that with `set_seed` our results are reproducing, since the README ones are non-trivially lower than the paper.

File: EC_finetune/senders.py (hunk 2 above)
Comment (anthonymirand) [reply EC_finetune/senders.py:194]: Makes sense! There's a lot of config overlap that makes this messy today but a good candidate to throw some inheritance into the mix in the future :-)

File: EC_finetune/__main__.py [hunk 3]
```diff
@@ -293,9 +293,38 @@ def main():
...
+    # point there won't be two competing sources of information to be merged.
+    # Today, we opt to pursue the latter route instead of fixing these config
+    # update bugs which occur throughout the pipeline.
+    # 
+    # As is, this code does not run. Please comment out the next line to allow
+    # the pipeline to run. Note that this removes all command line overrides.
     config_dict.update(args_dict)
     args_dict.update(config_dict)
 
+    set_seed(args.seed, args.n_gpu)
```
[the remark this answers is not in the document]
Comment (anthonymirand) [reply EC_finetune/__main__.py:326]: This fixes the reproducibility issues from the last PR. `set_seed` was dropped when the `add_adapters` was merged [here](https://github.com/CLMBRs/communication-translation/commit/ee81f7f22a6d1b5b6db174efde1dd7007812a76e#diff-58f85cba2e8b9e12750d38907a3ecd5af8560199c22690f77dde94ec1732d34f).

I would expect that T2I would still not reproduce 100% (since it never reproduced on the main branch) but this should get us back to the original I2I results based on my first-pass reproducibility results (assuming `add_adapters` did not introduce subtle side effects).

I'm leaning on not blocking this merge to validate these results (I2I/T2I both take 24+ hours to run). What do you think?

[the remark this answers is not in the document]
Comment (anthonymirand) [reply EC_finetune/__main__.py:326]: Also what README are you referring to? This PR description [link](https://github.com/CLMBRs/communication-translation/pull/13#issue-1380655635)?

[the remark this answers is not in the document]
Comment (shanest) [reply EC_finetune/__main__.py:326]: Sorry, yes, meant the PR description.  Agreed re not blocking!

File: EC_finetune/__main__.py (hunk 1 above)
Comment (anthonymirand) [reply EC_finetune/__main__.py:476]: Done!

File: EC_finetune/__main__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (anthonymirand) [reply EC_finetune/__main__.py:326]: So it turns out that I ran the T2I numbers after the seed was added back so our numbers do not change after a sanity run, and therefore are not reproducing here. This is kinda expected since Agatha was not able to have T2I reproduce when it was first written either.

[the remark this answers is not in the document]
Comment (shanest) [reply EC_finetune/__main__.py:326]: Great! My question was whether our new numbers for T2I are reproducing, not whether we can get the preprint's.  It sounds like they are, which is great!  In other words, we need reproducibility looking forward, not looking backward.
